pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -789,74 +789,70 @@ public static boolean isInternalTopic(String topic) {
 
     @VisibleForTesting
     void playLeader() {
-        if (role != Leader) {
-            log.info("This broker:{} is changing the role from {} to {}",
-                    pulsar.getLookupServiceAddress(), role, Leader);
-            int retry = 0;
-            while (true) {
+        log.info("This broker:{} is setting the role from {} to {}",
+                pulsar.getLookupServiceAddress(), role, Leader);
+        int retry = 0;
+        while (true) {
```

Comment:
Just wondering about a possible shutdown scenario. For those cases, it would be better to terminate loops based on the thread's interrupt status instead of having `while(true) {` loops.
```
        while (!Thread.currentThread.isInterrupted()) {
```

In addition to this, I'd recommend preserving Thread interrupt status whenever InterruptedException is caught.
```
try {
   Thread.sleep(x);
} catch (InterruptedException e) {
    // preserve thread's interrupt status
    Thread.currentThread().interrupt();
}
```
(one of the references https://stackoverflow.com/a/52623479)

Reply:
Thanks for sharing this practice. Updated.